### PR TITLE
Create DeflatedSharpeRatio.cs

### PR DESCRIPTION
**Pull request by Jay-Jay-D**
Implementation from the source paper: [The Deflated Sharpe Ratio: Correcting for Selection Bias, Backtest Overfitting and Non-Normality](https://papers.ssrn.com/sol3/papers.cfm?abstract_id=2460551).

### CODE REVIEW

**Comment by jameschch** [comment]
Thanks for fixing the formulas: this is outstanding. As discussed there was some amount of restructuring of my original gist to use as fitness. This causes problems with merging your fixes. Will do what I can to save you making too many more commits.

File: Optimization.Tests/DeflatedSharpeRatioFitnessTest.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+
+        [Test]
+        public void CalculateExpectedMaximumTest()
+        {
+            var unit = new DeflatedSharpeRatioWrapper();
+            unit.Initialize(null);
+            var actual = unit.CalculateExpectedMaximum();
+
+            Assert.AreEqual(4.2894064662728244, actual, 0.0000001);
```
Comment (jameschch) [inline Optimization.Tests/DeflatedSharpeRatioFitnessTest.cs]: We're expecting the expected maximum to return the value from the example formula: ~0.1132 as this is fed directly into the deflation. The code here is consistent (I think) with the python but not with the formula. One is mu+sigma*(...) wheres the formula has sqrt(1/2.25.)

File: Optimization.Tests/DeflatedSharpeRatioFitnessTest.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Optimization.Tests
+{
+
+    [TestFixture]
+    public class DeflatedSharpeRatioFitnessTest
```
Comment (jameschch) [inline Optimization.Tests/DeflatedSharpeRatioFitnessTest.cs]: This file will need to be renamed

File: Optimization/DeflatedSharpeRatio.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using MathNet.Numerics.Statistics;
+using MathNet.Numerics;
+using MathNet.Numerics.Distributions;
+using System.Runtime.CompilerServices;
+
+namespace Optimization
+{
+    public class DeflatedSharpeRatioFitness
```
Comment (jameschch) [inline Optimization/DeflatedSharpeRatio.cs]: This file will need to be renamed

File: Optimization/DeflatedSharpeRatio.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        {
+            return Normal.InvCDF(0, 1, x);
+        }
+
+        public double CalculateExpectedMaximum()
+        {
+            var asd = ZInverse(1 - 1 / N);
+            var qwe = ZInverse(1 - 1 / (N * Constants.E));
+            var maxZ = (1 - Constants.EulerMascheroni) * ZInverse(1 - 1 / N) + Constants.EulerMascheroni * ZInverse(1 - 1 / (N * Constants.E));
+            return SharpeRatio + Math.Sqrt(V) * maxZ;
```
Comment (jameschch) [inline Optimization/DeflatedSharpeRatio.cs]: see comment above regarding python vs. formula

File: Optimization.Tests/DeflatedSharpeRatioFitnessTest.cs (hunk 1 above)
Comment (Jay-Jay-D) [inline Optimization.Tests/DeflatedSharpeRatioFitnessTest.cs]: ~0.1132 is not the expected maximum, is the SR_0, if you check the equation (1) in page 7, the Expected maximum is the mean SR plus the annualized SR_0

Comment (jameschch) [inline Optimization.Tests/DeflatedSharpeRatioFitnessTest.cs]: In latest commit to master have made changes that allow recreation of the formulas.

**Pull request closed without merging**